File: mkchal/templates/pwn/sample.c
#include <stdio.h>

void setup() {{
    setbuf(stdin, NULL);
    setbuf(stdout, NULL);
    setbuf(stderr, NULL);
}}

// Sample challenge which just prints the flag free to delete this
int main() {{
    setup();

    char flag_buf[128] = {{ 0 }};

    FILE *file = fopen("./flag.txt", "r");
    if (file != NULL) {{
        fread(flag_buf, sizeof(char), sizeof(flag_buf), file);
    }}

    printf("Hello I am challenge: {name} and my flag is %s\n", flag_buf);
}}

File: mkchal/templates/web/sample.py
from flask import Flask
# Sample web flask challenge which serves the flag feel free to delete this

app = Flask(__name__)

@app.route("/")
def index():
    with open("./flag.txt", "r") as f:
        file = f.read()
    return "Hello I am challenge: {name} and my flag is " + file

if __name__ == "__main__":
    app.run("0.0.0.0", {port})